test/NuGet.Core.Tests/NuGet.Protocol.Tests/ServiceIndexResourceV3Tests.cs

```diff
@@ -24,6 +25,52 @@ public void Constructor_InitializesProperties()
             Assert.Equal("http://unit.test/b", resource.Entries[0].Uri.ToString());
         }
 
+        [Fact]
+        public void GetServiceEntries_WithResourceEndPoint_ThrowsException()
```

Comment:
The key is that insecureConnections is false and the sources are HTTP.

```suggestion
        public void GetServiceEntries_WithInsecureHttpResourceEndPoint_doNotAllowInsecureConnections_ThrowsException()
```